src/services/api/api.type.ts

```diff
@@ -0,0 +1,46 @@
+import { AxiosError } from 'axios';
+
+export class CustomError extends Error {
+  constructor(message: string) {
+    super(message);
+    Object.setPrototypeOf(this, CustomError.prototype);
+  }
+
+  public static fromAxiosError(error: unknown) {
+    const err = error as AxiosError;
+
+    if (!err.isAxiosError) {
+      return new CustomError(
+        'Sorry something went wrong from our side, please try again'
+      );
+    }
+    if (err.code === 'ERR_NETWORK') {
+      return new CustomError(
+        'Sorry something went wrong from our side, please try again'
```

Comment:
@siroshan Yo I didn't see the filename first—let's move this inside `api.ts` file or it's own `customError.ts` file inside `/api` since it's not a type definition